File: game/cave/climbing.py
import pygame

from animation import Animation
from audio_helper import AudioHelper
from cave.cave_resources import CaveResources
from cave.waiting_input import WaitingInput
from phone_events import PhoneEvents
from state import State


class Climbing(State):
    def process_events(self, phone_events: PhoneEvents):
        if self.get_frame_index() >= len(CaveResources.climbs):
            return WaitingInput

        if self.one_shot(1, "NuKommerJeg"):
            AudioHelper.play(CaveResources.nu_kommer_jeg, self.context.country)

        if self.one_shot(2, "PreFanfare"):
            AudioHelper.play(CaveResources.pre_fanfare, self.context.country)

        return None

    def render(self, screen):
        screen.blit(Animation.get_frame(CaveResources.climbs, self.get_frame_index()), (0, 0))

File: game-web/src/cave/climbing.ts
/** Port of game/cave/climbing.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { AudioHelper } from '../audioHelper';
import { CaveResources } from './caveResources';
import { WaitingInput } from './waitingInput';
import type { PhoneEvents } from '../phoneEvents';

export class Climbing extends State {
  processEvents(_events: PhoneEvents): StateClass | null {
    if (this.getFrameIndex() >= CaveResources.climbs.length) return WaitingInput;
    if (this.oneShot(1, 'NuKommerJeg')) AudioHelper.play(CaveResources.nuKommerJeg);
    if (this.oneShot(2, 'PreFanfare'))  AudioHelper.play(CaveResources.preFanfare);
    return null;
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getFrame(CaveResources.climbs, this.getFrameIndex()), 0, 0);
  }
}
